optimizely/odp/odp_config.py

```diff
@@ -0,0 +1,96 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Optional
+from threading import Lock
+
+
+class OdpConfig:
+    """
+    Contains configuration used for ODP integration.
+
+    Args:
+        api_host: The host URL for the ODP audience segments API (optional).
+        api_key: The public API key for the ODP account from which the audience segments will be fetched (optional).
+        segments_to_check: A list of all ODP segments used in the current datafile
+                            (associated with api_host/api_key).
+    """
+    def __init__(
+        self,
+        api_key: Optional[str] = None,
+        api_host: Optional[str] = None,
+        segments_to_check: Optional[list[str]] = None
+    ) -> None:
+        self.api_key = api_key
+        self.api_host = api_host
+        self.segments_to_check = segments_to_check or []
+        self.lock = Lock()
+
+    def update(self, api_key: Optional[str], api_host: Optional[str], segments_to_check: list[str]) -> bool:
+        """
+        Override the ODP configuration.
+
+        Args:
+            api_host: The host URL for the ODP audience segments API (optional).
+            api_key: The public API key for the ODP account from which the audience segments will be fetched (optional).
+            segments_to_check: A list of all ODP segments used in the current datafile
+                               (associated with api_host/api_key).
+
+        Returns:
+            True if the provided values were different than the existing values.
+        """
+        updated = False
+        with self.lock:
+            if self.api_key != api_key or self.api_host != api_host or self.segments_to_check != segments_to_check:
+                self.api_key = api_key
+                self.api_host = api_host
+                self.segments_to_check = segments_to_check
+                updated = True
+
+        return updated
+
+    def get_api_host(self) -> Optional[str]:
+        with self.lock:
+            return self.api_host
+
+    def set_api_host(self, api_host: Optional[str]) -> None:
+        with self.lock:
+            self.api_host = api_host
+
+    def get_api_key(self) -> Optional[str]:
+        with self.lock:
+            return self.api_key
+
+    def set_api_key(self, api_key: Optional[str]) -> None:
+        with self.lock:
+            self.api_key = api_key
+
+    def get_segments_to_check(self) -> Optional[list[str]]:
+        with self.lock:
+            return self.segments_to_check.copy()
+
+    def set_segments_to_check(self, segments_to_check: list[str]) -> None:
+        """
+        Replace qualified segments with provided list of segments.
+        Args:
+          segments_to_check: A list of segment names.
+        """
+        with self.lock:
+            self.segments_to_check = segments_to_check.copy()
```

Comment:
I do not see any case to set these props separately other than updating altogether with update(). Any use case in mind?

Reply:
Nope! Removed.